### ISSUE TITLE
Add "Guides" folder into docs/ to provide storys on how to do certain things

### ISSUE THREAD

**Issue by andygrunwald** [lansuite/lansuite#123]
## Context

Right now it is hard to reach a certain goal with LanSuite if you are not deep into the system.
Typical goals are:

* How to setup a docker env with the latest verison of LanSuite with the Database of my existing one?
* How do i test this System quickly?
* How to test the LanSuite system against multiple versions of MySQL?
* How to write an own module?
* How to write an own design?

and so on. I think there are multiple guides that can be written

The idea:
Create a `docs/guides/` folder.
Every guide is represented by one Markdown file like `test-lansuite-with-your-live-database.md`.
In the `README.md` of `docs/guides/` all guides are linked.

For this the API doc generation by php documentator needs to be passed to `docs/api/` or `docs/generated`.

**Comment by M4LuZ** [comment]
Hmm, generally a good idea, but should that be a docs file or should we just put that in the in the project wiki here on GitHub? At least editing-wise the later would be better.

Also the first entry should be:
* How to migrate from Release 4.2 to the GitHub master branch?

**Comment by M4LuZ** [comment]
Spontaneously started documenting.
https://github.com/lansuite/lansuite/wiki/HowTo%3A-Upgrade-from-Release-4.2
Is there maybe a way to export from the wiki to markdown files for a release?

**Comment by andygrunwald** [comment]
The wiki is also a valid approach.
But i would still prefer having them next to the source code, because

### Changing the product

The guides will change, when we do a change in the product.
This means we can adjust the guides in the same PR as we adjust the code.
This also offers us the possibility to search in our code and find this (e.g. to search after docker or mysql or so).
I made the experience from several projects, when the guides / docs are in another place (like the wiki) it is often forgotten to adjust them once the product changes.
Of course, you can create a checklist for a release and add a point like "Adjust documentation", but this is a big burden. Because at this point in time you have to go through all the changes and adjust the docs. This is not fun and no one wants to do this. The time is limited anyway.
When we have them next to the code, for every change the guide will be adjusted. This big tasks of adjusted the docs are splitted into thousands small parts which make it not a burden anymore.

### Available in a release

The docs folder is part of the release.
Means in every release the guides are shipped automatically with.
We don't have to introduce an extra "compile" / "export" step to make the release process "more" complicated. When we put this into the wiki we would need to introduce some sort of action here

### Available offline

When you are offline, you have all guides with you, like the source

### Possibility to publish them

Github Pages offers the generation of a side from a folder (e.g. docs). See [Configuring a publishing source for GitHub Pages](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/).
This would offers the benefit that we have them automatically updated, once we launch a LanSuite github page.
See #8 

What do you think @M4LuZ ?

**Comment by andygrunwald** [comment]
To publish those guides a nice idea is used in [golang/dep](https://github.com/golang/dep).

> This site, which is automatically generated from the [docs directory](https://github.com/golang/dep/tree/master/docs) within the dep repository by [docusaurus](http://docusaurus.io/), is now that comprehensive source of docs. More so than any individual bit of information, it provides some broader benefits:
... [continue reading at the link]

See [Announcing dep v0.4.1 (with docs!)](https://golang.github.io/dep/blog/2018/01/23/announce-v0.4.0.html) for details.

**Comment by M4LuZ** [comment]
Just noted that the GitHub Wiki is just based on another Git repo per project.
So if we add that as submodule for `docs` we should have the best of both worlds.
We can include document changes in commits, have autogenerated visible pages and have to possibility of direct editing on GitHub.
At least that is my understanding, found the following while having a quick look at https://gist.github.com/subfuzion/0d3f19c4f780a7d75ba2

**Comment by andygrunwald** [comment]
Yes, the Wiki is just another Git Repo. To use the wiki in a full manner like we can use the guides as simple markdown files would have some disadvantages/challenges:

#### Git submodules

This idea sounds nice at first, but this brings a new world of challenges. See
* [Git: why submodules are evil](http://slopjong.de/2013/06/04/git-why-submodules-are-evil/)
* [Two common problems with Git submodules](https://rufflewind.com/2017-02-21/quirks-with-submodules)
* [Why your company shouldn’t use Git submodules](https://codingkilledthecat.wordpress.com/2012/04/28/why-your-company-shouldnt-use-git-submodules/)

Next to this, for every new guide, the submodule pointer in this repository needs to be raised.
To write a guide, you need to a) write it and b) raise a submodule pointer (which is not quite clear when you write a guide that you need to do this)

#### Peer reviews

With Markdown files, you treat your guides like source code. And like source code, you do peer reviews via Pull Requests for every guide. In theory (and practice!) this should raise the quality of the guides.
Technically this can be done also with the wiki (see the link you have provided), but in practice, no one will follow this guide, when Github offers a nice WYSIWYG UI.

#### Automatically updates your Github page

Having the guides as markdown files would offer the possibility to display them on the GitHub page.
Every commit to the guides will be then automatically updated on our page. See [Announcing dep v0.4.1 (with docs!)](https://golang.github.io/dep/blog/2018/01/23/announce-v0.4.0.html) as an example.

This is not possible with the Github Wiki out of the box. You would need to write a logic/script for it and let it run as a cronjob.

**Comment by andygrunwald** [comment]
The base for the new documentation was set in #313. Thanks for merging @M4LuZ!
I published this site on https://lansuite.github.io/lansuite/

What is left TODO:
- [ ] Document on how to publishing the site
- [ ] Document on how to write a new documentation article
- [ ] Document on how to write a new blog post
- [ ] Adjusting the `README-DE.md` like the current `README.md`
- [ ] Document on how to do a LANSuite release
- [ ] Add the site generation to the release process
- [ ] Add more context about the single configuration settings into the documentation for the `Configuration` chapter
- [ ] Finish the article `Upgrade to a newer version`
- [ ] Rework the article `Generating API docs`
- [ ] Document on how users can test the Coding Guidelines in the `Coding style guide` chapter

Anything missing from your side @M4LuZ ?
